### PR TITLE
refac: Use a single HTTP client throughout the HTTPEventDispatcher life-cycle

### PR DESCRIPTION
**Pull request by pawels-optimizely**

### CODE REVIEW

File: pkg/event/dispatcher.go [hunk 1]
```diff
@@ -41,13 +41,13 @@ type Dispatcher interface {
 
 // HTTPEventDispatcher is the HTTP implementation of the Dispatcher interface
 type HTTPEventDispatcher struct {
+	requester *utils.HTTPRequester
 }
 
 // DispatchEvent dispatches event with callback
-func (*HTTPEventDispatcher) DispatchEvent(event LogEvent) (bool, error) {
+func (eventDispacher *HTTPEventDispatcher) DispatchEvent(event LogEvent) (bool, error) {
```
Comment (mikeng13) [inline pkg/event/dispatcher.go]: nit: should this be shortened to something like `ed` like we do for other func receivers?

File: pkg/event/processor_test.go [hunk 2]
```diff
@@ -506,7 +504,7 @@ func BenchmarkWithQueueSize(b *testing.B) {
 	for _, merge := range merges {
 		var totalSent = 0
 		var numberRun = 0
-		b.Run(merge.name, func (b *testing.B) {
+		b.Run(merge.name, func(b *testing.B) {
```
Comment (mikeng13) [inline pkg/event/processor_test.go]: I think this should've been caught by our linters. Can we update them to catch these in a follow-up PR?

Comment (pawels-optimizely) [reply pkg/event/processor_test.go]: this is IDE's thing,  there is difference between Goland and VSC, I will look into that

File: pkg/event/dispatcher.go (hunk 1 above)
Comment (pawels-optimizely) [reply pkg/event/dispatcher.go]: addressed

**Pull request merged**